platforms/core-runtime/launcher/src/main/java/org/gradle/launcher/cli/DefaultCommandLineActionFactory.java

```diff
@@ -192,38 +201,103 @@ public void execute(ExecutionListener executionListener) {
     }
 
     public static class ShowVersionAction implements Action<ExecutionListener> {
+        private final Supplier<Parameters> parametersSupplier;
+
+        public ShowVersionAction(Supplier<Parameters> parametersSupplier) {
+            this.parametersSupplier = parametersSupplier;
+        }
+
         @Override
         public void execute(ExecutionListener executionListener) {
+            Parameters parameters = parametersSupplier.get();
             DefaultGradleVersion currentVersion = DefaultGradleVersion.current();
 
             System.out.println();
             System.out.println("------------------------------------------------------------");
             System.out.println("Gradle " + currentVersion.getVersion());
             System.out.println("------------------------------------------------------------");
             System.out.println();
-            System.out.println("Build time:   " + currentVersion.getBuildTimestamp());
-            System.out.println("Revision:     " + currentVersion.getGitRevision());
-            System.out.println();
-            System.out.println("Kotlin:       " + KotlinDslVersion.current().getKotlinVersion());
-            System.out.println("Groovy:       " + ReleaseInfo.getVersion());
-            System.out.println("Ant:          " + Main.getAntVersion());
-            System.out.println("JVM:          " + Jvm.current());
-            System.out.println("OS:           " + OperatingSystem.current());
-            System.out.println();
+            printAligned(ImmutableList.of(
+                new Line.KeyValue("Build time", currentVersion.getBuildTimestamp()),
+                new Line.KeyValue("Revision", currentVersion.getGitRevision()),
+                new Line.Blank(),
+                new Line.KeyValue("Kotlin", KotlinDslVersion.current().getKotlinVersion()),
+                new Line.KeyValue("Groovy", ReleaseInfo.getVersion()),
+                new Line.KeyValue("Ant", Main.getAntVersion()),
+                new Line.KeyValue("Launcher JVM", Jvm.current().toString()),
+                new Line.KeyValue("Requested Daemon JVM", parameters.getDaemonParameters().getRequestedJvmCriteria().toString()),
```

Comment:
🤔  Now that I've seen some of the output... a couple of tweaks:
- Maybe `Daemon JVM` instead of `Requested Daemon JVM`?
- `(no JDK specified, using Launcher JVM)` -> nothing.  If we say "using Launcher JVM" it sounds like we might be reusing the same process.
- `(from -Dorg.gradle.java.home)` -> `(from org.gradle.java.home)`.  You can specify org.gradle.java.home in multiple ways.
- `(Daemon JVM criteria from gradle/gradle-daemon-jvm.properties)` -> `(from gradle/gradle-daemon-jvm.properties)`